File: my-folder/1894-merge-strings-alternately/solution.java
class Solution {
    public String mergeAlternately(String word1, String word2) {
        StringBuilder sb = new StringBuilder();
        int min = Math.min(word1.length(), word2.length());
        for (int i = 0; i < min; i++) {
            sb.append(word1.charAt(i));
            sb.append(word2.charAt(i));
        }
        if (min < word1.length()) {
            sb.append(word1.substring(min, word1.length()));
        }
        else if (min < word2.length()){
            sb.append(word2.substring(min, word2.length()));
        }
        return sb.toString();
    }
}

File: my-folder/1894-merge-strings-alternately/solution.py
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        i, j = 0, 0
        res = ""
        while i < len(word1) and j < len(word2):
            res += word1[i]
            res += word2[j]
            i += 1
            j += 1
        res += word1[i:len(word1)]
        res += word2[j:len(word2)]
        return res

